### PR TITLE
Reorganize docs and some fixes

### LINKED ISSUE #33: add back docs after switch to MDA theme
PR #31 temporarily removed docs from `api.rst`
```
.. automodule:: mdahole2.analysis.utils
   :members:

.. automodule:: mdahole2.analysis.templates
   :members:
```
because it does not work nicely with the MDA sphinx theme.

These docs need to be added back.

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #33

Changes made in this Pull Request:

 - Adds docs back, fixes bibliography, organises to make a bit more sense, removes unused things, minor fixes to formatting, links, and otherwise.

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+The class can be set-up and run like a normal MDAnalysis analysis class::
+
+    import MDAnalysis as mda
+    from MDAnalysis.tests.datafiles import MULTIPDB_HOLE
+    from MDAnalysis.analysis import hole2
+
+    u = mda.Universe(MULTIPDB_HOLE)
+
+    ha = hole2.HoleAnalysis(u, executable='~/hole2/exe/hole') as h2:
```
Comment (IAlibay) [inline docs/source/usage.rst]: Here and above/below, our recommended approach here is to install hole2 via conda-forge. There doesn't seem to be any instructions for users on how to deal with the CF installation case (i.e. they may not easily know the path to their installation)

Comment (lilyminium) [reply docs/source/usage.rst]: Thanks for that note Irfan, I've added a section on finding the path. I'll update installation notes when there's a full c-f package.

Comment (lilyminium) [reply docs/source/usage.rst]: Actually, doesn't hurt to update now, especially if the order of events has to go release -> package...

Comment (lilyminium) [reply docs/source/usage.rst]: Added some very brief installation notes as well!

File: docs/source/usage.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+.. code-block:: bash
+
+    $ conda activate mdahole2
+    $ which hole
+    /home/username/miniforge3/envs/mdahole2/bin/hole
+
+If you have installed HOLE from the original source, the executable
+may be in your home directory instead, e.g. a path such as ``~/hole2/exe/hole``.
+
+Throughout the rest of this documentation, we will assume that your executable
```
Comment (IAlibay) [inline docs/source/usage.rst:30]: Sorry I think I failed to communicate part of my initial question properly.

Our tests, and I would guess also the expected usage from our users, is that `hole` is found within $PATH. That is to say, it seems like our intended use case is that `executable` isn't explicitly passed by default? (please do correct me if that is an incorrect view of how we expect folks to do things).

In that case, I wonder if the wording here might either: a) tell folks to pass `executable` if they have hole defined in a non-standard way and then not show examples using `executable` or b) tell folks that you can just not pass `executable` in most cases and it'll work properly?

Comment (lilyminium) [reply docs/source/usage.rst:30]: Right, I see what you mean, thanks for the clarification. I've updated the docs as such.

File: mdahole2/analysis/__init__.py [hunk 3]
```diff
@@ -30,280 +30,6 @@
 
 .. versionadded:: 1.0.0
```
Comment (IAlibay) [inline mdahole2/analysis/__init__.py]: Should we remove this too? It looks like this is a holdover from the core library.

File: mdahole2/analysis/hole.py [hunk 4]
```diff
@@ -212,6 +212,19 @@ def hole(pdbfile,
 
     .. versionadded:: 1.0
```
Comment (IAlibay) [inline mdahole2/analysis/hole.py]: And here too, this probably should go too?

**Review by IAlibay — APPROVED** [review]
Thanks @lilyminium - overall lgtm, although I'm seeing these leftover versionadded (probably copied over from the core library). Should we remove them too?

File: mdahole2/analysis/hole.py (hunk 4 above)
Comment (lilyminium) [reply mdahole2/analysis/hole.py]: Sure, I've removed those. there's some deprecation wording that can probably be updated in a different PR.

**Pull request merged**